### PR TITLE
Added <kbd> tags where appropriate

### LINKED ISSUE #1500: Mark up keyboard keys with `<kbd>` per Google Style
Per [writing guidelines](https://developers.google.com/style/ui-elements#keyboard).

### PR DESCRIPTION
**Pull request by atsansone**
Fixes #1500

### CODE REVIEW

File: src/tools/devtools/release-notes/release-notes-2.8.0-src.md
Comment (parlough) [inline src/tools/devtools/release-notes/release-notes-2.8.0-src.md:1]: I haven't looked at this PR yet, but please revert the changes here. DevTools release notes are rendered in DevTools so they have to be pure Markdown (can't have inline HTML). Thanks!

File: src/cookbook/testing/unit/introduction.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ this is the convention used by the test runner when searching for tests.
 
 When you're finished, the folder structure should look like this:
 
-```
+```terminal
```
Comment (parlough) [inline src/cookbook/testing/unit/introduction.md:59]: This isn't terminal code. It's a representation of a file structure depicted in a code block for consistency spacing/formatting.

Simply specifying `nocode` is probably sufficient and is what we do for similar situations on dart.dev.

````suggestion
```nocode
````

**Review by parlough — COMMENTED** [review]
I haven't looked closely at this PR yet, but two comments from a really quick scan:

File: src/_includes/docs/debug/debug-flow-vscode-as-start.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
    **View** <span aria-label="and then">></span>
    **Command Palette...**
 
-   You can also press <kbd>Command</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>.
+   You can also press <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>.
```
Comment (parlough) [inline src/_includes/docs/debug/debug-flow-vscode-as-start.md:14]: I know you didn't touch this here, but looks like this should also have a non-macos key binding.

File: src/platform-integration/windows/building.md [hunk 3]
```diff
@@ -167,10 +167,15 @@ Follow these steps:
...
-   to generate the necessary dependencies.
-   You should now be able to run **Debug / Start Debugging**
-   (or press F5) to run the Windows app from Visual Studio.
+1. To generate the necessary dependencies,
+   run **Build** > **Build Solution**
+
+   You can also press/
+   <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>B</kbd>.
+
+   To run the Windows app from Visual Studio, go to **Debug** > **Start Debugging**.
```
Comment (parlough) [inline src/platform-integration/windows/building.md:176]:
```suggestion
   To run the Windows app from Visual Studio,
   go to **Debug** > **Start Debugging**.
```

File: src/platform-integration/windows/building.md [hunk 4]
```diff
@@ -167,10 +167,15 @@ Follow these steps:
...
    and choose **Set as Startup Project**.
 
-1. Run **Build / Build Solution** (or press Ctrl+Shift+B)
-   to generate the necessary dependencies.
-   You should now be able to run **Debug / Start Debugging**
-   (or press F5) to run the Windows app from Visual Studio.
+1. To generate the necessary dependencies,
+   run **Build** > **Build Solution**
+
+   You can also press/
```
Comment (parlough) [inline src/platform-integration/windows/building.md:173]: Did you intend to leave this slash?

```suggestion
   You can also press
```

File: src/tools/vs-code.md [hunk 5]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
- 1. Select a **Project location**.
- 1. Enter your desired **Project name**.
+1. Open the Command Palette.
+
+   - On Windows or Linux,
+     press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>P</kbd>.
+   - On macOS,
+     press <kbd>Cmd</kbd> + <kbd>Shift</kbd>+<kbd>P<kbd>.
+
+1. Tyoe `flutter`.
```
Comment (parlough) [inline src/tools/vs-code.md:51]:
```suggestion
1. Type `flutter`.
```

File: src/tools/vs-code.md [hunk 6]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
+1. Select the **Flutter: New Project**.
```
Comment (parlough) [inline src/tools/vs-code.md:52]:
```suggestion
1. Select the **Flutter: New Project** command.
```

File: src/tools/vs-code.md [hunk 7]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
 
 ### Opening a project from existing source code
 
 To open an existing Flutter project:
 
- 1. Click **File > Open** from the main IDE window.
- 1. Browse to the directory holding your existing
-    Flutter source code files.
- 1. Click **Open**.
+1. From the main IDE window, go to **File** > **Open**.
```
Comment (parlough) [inline src/tools/vs-code.md:63]: I know this is pre-existing, but what does "main IDE window mean"? There is only one window in VS code, so it may have been included accidentally.

Perhaps "In the VS Code menu bar ..."?

File: src/tools/vs-code.md [hunk 8]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
 enables the following:
 
-* Syntax highlighting
-* Code completions based on rich type analysis
-* Navigating to type declarations
-  (**Go to Definition** or `F12`),
+- Syntax highlighting
+- Code completions based on rich type analysis
+- Navigating to type declarations
+  (**Go to Definition** or press <kbd>F12</kbd>),
```
Comment (parlough) [inline src/tools/vs-code.md:76]: The "Go to Definition" doesn't have an action. Either add one for that or drop it from the key binding.

File: src/tools/vs-code.md [hunk 9]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
   and finding type usages
-  (**Find All References** or `Shift`+`F12`)
-* Viewing all current source code problems
-  (**View > Problems** or `Ctrl`+`Shift`+`M`
-  (`Cmd`+`Shift`+`M` on macOS))
+  (**Find All References** or press <kbd>Shift</kbd> + <kbd>F12</kbd>)
```
Comment (parlough) [inline src/tools/vs-code.md:78]: The "Find All References" doesn't have an action. Either add one for that or drop it from the key binding.

File: src/tools/vs-code.md [hunk 10]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
+- Viewing all current source code problems
+  (**View** > **Problems**
+  or <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd>
+  or <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> on macOS)
```
Comment (parlough) [inline src/tools/vs-code.md:82]: The "View > Problems" doesn't have an action. Either add one for that or drop it from the key binding.

File: src/tools/vs-code.md [hunk 11]
```diff
@@ -101,14 +110,16 @@ device name (or the message **No Devices**):<br>
...
+
+- If you do not see a Flutter version number or device info,
+  your project might not have been detected as a Flutter project.
+  Ensure that the folder that contains your `pubspec.yaml` is
+  inside a VS Code **Workspace Folder**.
+- If the status bar reads **No Devices**, Flutter has not been
+  able to discover any connected iOS or Android devices or simulators.
+  You need to connect a device, or start a simulator or emulator,
+  to proceed.
+
```
Comment (parlough) [inline src/tools/vs-code.md:122]:
```suggestion
  - If you do not see a Flutter version number or device info,
    your project might not have been detected as a Flutter project.
    Ensure that the folder that contains your `pubspec.yaml` is
    inside a VS Code **Workspace Folder**.
  - If the status bar reads **No Devices**, Flutter has not been
    able to discover any connected iOS or Android devices or simulators.
    You need to connect a device, or start a simulator or emulator,
    to proceed.
```

File: src/tools/vs-code.md [hunk 12]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
+  tools that run in a browser.
+- Using VS Code's built-in debugging features,
+  such as setting breakpoints.
 
-  The instructions below describe features available in VS Code.
-  For information on using launching DevTools, see
-  [Running DevTools from VS Code][] in the [DevTools][] docs.
+The instructions below describe features available in VS Code.
+For information on using launching DevTools, see
+[Running DevTools from VS Code][] in the [DevTools][] docs.
```
Comment (parlough) [inline src/tools/vs-code.md:98]: Our current style is to indent alerts two spaces.

File: src/tools/vs-code.md [hunk 13]
```diff
@@ -128,49 +139,54 @@ running or debugging.
 
 ### Run app without breakpoints
 
- 1. Click **Run > Start Without Debugging** in the
-    main IDE window, or press `Ctrl`+`F5`.
+Go to **Run** > **Start Without Debugging** in the main IDE window.
```
Comment (parlough) [inline src/tools/vs-code.md:142]: Same here. Consider dropping or replacing "main IDE window".

File: src/tools/vs-code.md [hunk 14]
```diff
@@ -128,49 +139,54 @@ running or debugging.
...
-         "type": "dart",
-         "flutterMode": "debug"
-       }
-     ]
-     ```
- 1. Run the app through the **Run** view. 
+1. Open the `launch.json` file in VS Code.
+
+   If you do not have a `launch.json` file, go to
+   the **Run** view in VS Code and click **create a launch.json file**.
```
Comment (parlough) [inline src/tools/vs-code.md:169]: This might need updating. I wasn't sure what **Run** view is. Perhaps "**Run and Debug** tag"?

File: src/tools/vs-code.md [hunk 15]
```diff
@@ -251,14 +270,13 @@ can assist in correcting it.
 Snippets can be used to speed up entering typical code structures.
 They are invoked by typing their prefix,
 and then selecting from the code completion window:
-![Snippets]({{site.url}}/assets/images/docs/tools/vs-code/snippets.png){:width="100%"}
+![Snippets]({{site.url}}/assets/images/docs/tools/vs-code/snippets.png){:width="100%"}-
+-he Flutter extension includes the following -nippets:
```
Comment (parlough) [inline src/tools/vs-code.md:275]: Not sure what happened here 😄 

```suggestion
![Snippets]({{site.url}}/assets/images/docs/tools/vs-code/snippets.png){:width="100%"}

The Flutter extension includes the following snippets:

```

File: src/tools/vs-code.md [hunk 16]
```diff
@@ -251,14 +270,13 @@ can assist in correcting it.
...
 
-The Flutter extension includes the following snippets:
-
-* Prefix `stless`: Create a new subclass of `StatelessWidget`.
-* Prefix `stful`: Create a new subclass of `StatefulWidget`
+- Prefix `stless`: Create a new subclass of -StatelessWidget`.
```
Comment (parlough) [inline src/tools/vs-code.md:276]:
```suggestion
- Prefix `stless`: Create a new subclass of `StatelessWidget`.
```

File: src/ui/interactivity/focus.md [hunk 17]
```diff
@@ -30,36 +30,36 @@ creating `FocusNode` objects](#best-practices-for-creating-focusnode-objects).
...
+  the primary focus node and its ancestors.
+- **Focus chain** - An ordered list of focus nodes that starts at the primary
+  focus node and follows the branches of the focus tree to the root of the
+  focus tree.
+- **Focus scope** - A special focus node whose job is to contain a group of
+  other focus nodes, and allow only those nodes to receive focus. It contains
+  information about which nodes were previously focused in its subtree.
+- **Focus traversal** - The process of moving from one focusable node to
+  another in a predictable order. This is typically seen in applications when
+  the user presses the <kbd>Tab</kbd> to move to the next focusable control or
```
Comment (parlough) [inline src/ui/interactivity/focus.md:61]: Either drop the "the" or add back "key".

File: src/ui/interactivity/focus.md [hunk 18]
```diff
@@ -422,7 +422,7 @@ your custom controls.
 Once an application has the ability to focus, the next thing many apps want to
 do is to allow the user to control the focus using the keyboard or another input
 device. The most common example of this is "tab traversal" where the user
-presses the **Tab** key to go to the "next" control. Controlling what "next"
+presses the <kbd>Tab</kbd> to go to the "next" control. Controlling what "next"
```
Comment (parlough) [inline src/ui/interactivity/focus.md:425]: Either drop the "the" or add back "key".

**Review by parlough — APPROVED** [review]
Thanks for working on this!

I approved but please fix the following issues first:

File: src/tools/vs-code.md (hunk 12 above)
Comment (atsansone) [reply src/tools/vs-code.md:98]: Yes, and Markdownlint shows this as violating [MD007](https://github.com/DavidAnson/markdownlint/blob/main/doc/md007.md) because of the list. I dedented because of that warning.

File: src/cookbook/testing/unit/introduction.md [hunk 19]
```diff
@@ -156,31 +156,30 @@ The Flutter plugins for IntelliJ and VSCode support running tests.
...
-    3. Click the `Start Debugging` option
-    4. *Alternatively, use the appropriate keyboard shortcut
-       for your platform.*
+- **IntelliJ**
+
+  1. Open the `counter_test.dart` file
+  1. Go to **Run** > **Run 'tests in counter_test.dart'**.
+     You can also press the appropriate keyboard shortcut for your platform.
+
+- **VSCode**
```
Comment (sfshaza2) [inline src/cookbook/testing/unit/introduction.md:165]: Given our current situation, I generally swap the order and put VSCode first.

File: src/tools/vs-code.md [hunk 20]
```diff
@@ -25,12 +25,12 @@ Updates to the extensions are shipped on a regular basis.
 By default, VS Code automatically updates extensions when
 updates are available.
 
-To install updates manually:
+To install updates yourself:
```
Comment (sfshaza2) [inline src/tools/vs-code.md:28]: I'm curious. Why is "yourself" better than "manually"? It seems to me that manually is more precise.

File: src/tools/vs-code.md [hunk 21]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
+  or <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd>
+  or <kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> on macOS)
   Any analysis issues are shown in the Problems pane:<br>
   ![Problems pane]({{site.url}}/assets/images/docs/tools/vs-code/problems.png){:.mw-100.pt-1}
 
 ## Running and debugging
 
 {{site.alert.note}}
-  You can debug your app in a couple of ways.
+You can debug your app in a couple of ways.
```
Comment (sfshaza2) [inline src/tools/vs-code.md:89]: No, all site alert notes should be indented at least by two.

File: src/tools/vs-code.md [hunk 22]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
 
-  * Using [DevTools][], a suite of debugging and profiling
-    tools that run in a browser. 
-  * Using VS Code's built-in debugging features,
-    such as setting breakpoints.
+- Using [DevTools][], a suite of debugging and profiling
```
Comment (sfshaza2) [inline src/tools/vs-code.md:91]: Why is "-" better than asterisk? We generally use asterisk, which I prefer.

File: src/tools/vs-code.md [hunk 23]
```diff
@@ -41,56 +41,65 @@ There are a couple ways to create a new project.
...
+  tools that run in a browser.
+- Using VS Code's built-in debugging features,
+  such as setting breakpoints.
 
-  The instructions below describe features available in VS Code.
-  For information on using launching DevTools, see
-  [Running DevTools from VS Code][] in the [DevTools][] docs.
+The instructions below describe features available in VS Code.
+For information on using launching DevTools, see
+[Running DevTools from VS Code][] in the [DevTools][] docs.
```
Comment (sfshaza2) [inline src/tools/vs-code.md:98]: Please undo all of these changes to this alert.

File: src/tools/vs-code.md [hunk 24]
```diff
@@ -128,49 +139,54 @@ running or debugging.
...
+      {
+        "name": "Flutter",
+        "request": "launch",
+        "type": "dart",
+        "flutterMode": "debug"
+      }
+    ]
+    ```
+
+1. Run the app through the **Run** view.
```
Comment (sfshaza2) [inline src/tools/vs-code.md:189]: What's going on in this section with the red highlights?

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Thanks for reviewing it already, @parlough!

File: src/tools/vs-code.md (hunk 20 above)
Comment (atsansone) [reply src/tools/vs-code.md:28]: I find "automatically" and "manually" unnecessary. A user can't do anything automatically and no application or process can do something manually. Using "yourself" makes it clear who is performing the action. So, it's a double GSG win: active voice and remove extraneous words.

File: src/tools/vs-code.md (hunk 22 above)
Comment (atsansone) [reply src/tools/vs-code.md:91]: Markdownlint is flagging the list as indented, so it expected `-` for a second level list.

image

File: src/tools/vs-code.md (hunk 24 above)
Comment (atsansone) [reply src/tools/vs-code.md:189]: This section was moved, but not in this PR.
